app/controllers/api/v1/listen_signals_controller.rb

```diff
@@ -5,36 +5,32 @@ class Api::V1::ListenSignalsController < Api::V1::BaseController
   before_action :ensure_user_can_get_listen_signal, only: [:show]
 
   def update
-    @signal = current_user.brand.listen_signals.where(name: signal_params[:name]).first
+    @listen_signal = ListenSignal.find(params[:id])
+    signal_params = signal_params(params)
+    @listen_signal.update_attributes!(signal_params)
 
-    update_signal(@signal, signal_params)
-    update_responses(@signal, signal_params['firstResponse'], signal_params['repeatResponse'])
+    default_response = update_response(@listen_signal.default_response, params[:default_response])
+    repeat_response = update_response(@listen_signal.repeat_response, params[:repeat_response])
 
-    flash[:success] = "Alert:  #{@signal.name} signal updated!"
-    render json: {
-      signal: @signal.to_json,
-      responses: @signal.responses.to_json
-    }
+    if @listen_signal 
+      render json: @listen_signal, each_serializer: ListenSignalSerializer
+    end
   end
 
   def create
-    @brand = current_user.brand
-    @identity = @brand.identities.first
-
-    name = signal_params['name']
-    signal_type = signal_params['signalType']
-    first_response = signal_params['firstResponse']
-    repeat_response = signal_params['repeatResponse']
-    exp_date = signal_params['expirationDate']
-
-    @signal = ListenSignal.create_signal(@brand, @identity, name, signal_type, exp_date)
-    create_template_response(@signal, first_response, repeat_response, exp_date)
-
-    if @signal
-      render json: {
-        signal: @signal.to_json,
-        responses: @signal.responses.to_json
-      }
+    brand = current_user.brand
+    identity = brand.identities.first
+    signal_params = signal_params(params)
+    signal_params[:brand] = brand
+    signal_params[:identity] = identity
+    @listen_signal = ListenSignal.create!(signal_params)
+
+    default_response_msg = params[:default_response]
+    repeat_response_msg = params[:repeat_response]
+    create_grouped_response(default_response_msg, repeat_response_msg)
+
+    if @listen_signal
+      render json: @listen_signal, each_serializer: ListenSignalSerializer
     else
       flash[:error] = 'Alert:  Please fill out missing fields'
```

Comment:
> so in if/else statement here:

> can I just use the BaseController class ``` show_error(e) ```? 

> ```ruby
if @listen_signal
   blah blah
else
   e = ApiErrors::StandardError.new('signal failed bra', 'please you forgot shit', 400)
   render show_error(e)
end

```

This isn't really going to be a scenario because we are using `#create!` which raises errors. The `rescue_from` for `ActiveRecord::ActiveRecordError` will rescue that error. Within that rescue from, we can raise the `ApiErrors::StandardError` so that it gets sent to the client properly.